src/libraries/System.Net.Security/src/System/Net/Security/TlsFrameHelper.cs

```diff
@@ -179,27 +179,51 @@ public static bool TryGetFrameHeader(ReadOnlySpan<byte> frame, ref TlsFrameHeade
             {
                 header.Type = (TlsContentType)frame[0];
 
-                if (frame.Length >= 3)
+                if (frame.Length > 4)
                 {
                     // SSLv3, TLS or later
                     if (frame[1] == 3)
                     {
-                        if (frame.Length > 4)
-                        {
-                            header.Length = ((frame[3] << 8) | frame[4]);
-                        }
-
+                        header.Length = ((frame[3] << 8) | frame[4]);
                         header.Version = TlsMinorVersionToProtocol(frame[2]);
+                        return true;
                     }
-                    else
+                    else if (frame[2] == (byte)TlsHandshakeType.ClientHello &&
+                       frame[3] == 3) // SSL3 or above
                     {
-                        header.Length = -1;
-                        header.Version = SslProtocols.None;
+                        int length;
+                        if ((frame[0] & 0x80) != 0)
+                        {
+                            // Two bytes
+                            length = (((frame[0] & 0x7f) << 8) | frame[1]) + 2;
+                        }
+                        else
+                        {
+                            // Three bytes
+                            length = (((frame[0] & 0x3f) << 8) | frame[1]) + 3;
+                        }
+
+                        // max frame for SSLv2 is 32767.
+                        // However, we expect something reasonable for initial HELLO
+                        // We don't have enough logic to verify full validity,
+                        // the limits bellow are queses.
```

Comment:
```suggestion
                        // the limits bellow are guesses.
```

On the "1000", what happens if the length is >= 1000? I've seen some chunky ClientHellos in the past, largely because of a large cipher suite list and ClientHello extensions.

Reply:
This is basically legacy for the Ssl2 formatted ClientHello. The cases I saw with the IoT devices the cipher list has 16 items and there are no extensions by definition. This may be Windows-only - I'm not sure (yet) if OpenSSL even supports that. 
The 1000 is purely arbitrary. I was exited about https://github.com/dotnet/runtime/pull/64322. as it allowed to remove some old cruft but now I have second thought. This change alone does not make it work for Unified Hello.